src/checkers/inference/InferenceTreeAnnotator.java

```diff
@@ -114,13 +117,23 @@ public Void visitClass(final ClassTree classTree, final AnnotatedTypeMirror clas
         InferenceUtil.testArgument(classType instanceof AnnotatedDeclaredType,
                 "Unexpected type for ClassTree ( " + classTree + " ) AnnotatedTypeMirror ( " + classType + " ) ");
 
-        // For anonymous classes, we do not create additional variables, as they
-        // were already handled by the visitNewClass. This would otherwise result
-        // in new variables for an extends clause, which then cannot be inserted.
-        if (!InferenceUtil.isAnonymousClass(classTree)) {
-            this.variableAnnotator.visit(classType, classTree);
+        // Annotate the enclosing type if it exists
+        AnnotatedDeclaredType declType = (AnnotatedDeclaredType) classType;
+        AnnotatedDeclaredType enclosingType = declType.getEnclosingType();
+        TreePath classPath = atypeFactory.getPath(classTree);
+        if (classPath != null) {
+            ClassTree enclosingClass = TreeUtils.enclosingClass(classPath.getParentPath());
+            if (enclosingType != null && enclosingClass != null) {
+                variableAnnotator.visit(enclosingType, enclosingClass);
+            }
         }
 
+        // For anonymous classes, also fully annotate the classTree, which will be
```

Comment:
Please try clarifying this comment.
This now always annotates the classTree, so the beginning is confusing.
The 'Note' about what should be done is also not really understandable. Maybe explain this one in visitNewClass and here only have a reference to that explanation?